docs/concept.rst

```diff
@@ -116,3 +116,50 @@ directory. Below you can see node names for different files:
     +-------------------------------+-----------------------+
     | wget/download/smoke.fmf       | /download/smoke       |
     +-------------------------------+-----------------------+
+
+
+Identifiers
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+Node names are unique across the metadata tree and thus can be
+used as identifiers for local referencing. In order to reference
+remote fmf nodes a full ``fmf identifier`` is defined as a
+dictionary containing keys with the following meaning:
+
+url
+    Git repository containing the metadata tree. Use any format
+    acceptable by the ``git clone`` command.
+ref
+    Branch, tag or commit specifying the desired git revision.
+    This is used to perform a ``git checkout`` in the repository.
+    By default the ``master`` branch is used.
+path
+    Path from the git repository root to the metadata tree root.
+    By default ``/`` is used as very often the fmf metadata root
+    is aligned with the git repository root.
+name
+    Node name as defined by the hierarchy in the metadata tree.
+
+Here's a full fmf identifier example::
+
+    url: https://github.com/psss/fmf
+    ref: 0.10
+    path: /examples/wget
+    name: /download/test
+
+Use default values for ``ref`` and ``path`` to reference the
+latest version of the smoke plan from the ``master`` branch::
+
+    url: https://github.com/psss/fmf
+    name: /plans/smoke
+
+Let's freeze the stable test version by using a specific commit::
+
+    url: https://github.com/psss/fmf
+    ref: f24ef3f
+    name: /tests/basic/filter
+
+Reference another repository on the local filesystem::
+
+    url: file:///home/psss/git/tmt
+    name: /plans/smoke
```

Comment:
Does that one refer to `master` or to the working tree?

Reply:
Unless specified, `ref` defaults to master. So this would be `master`. Would you expect something else?

Reply:
good point ... and I do not have a good answer what it should be .... :(

Reply:
Me being myself, I may be puzzled why my uncommitted working tree changes are not picked up. I don't see a nice solution, but it should be probably explicitly stated that it will clone and checkout master, not use the working tree.

Reply:
We could also make this example a little bit more clear and specific by adding `ref: HEAD` and stating it would use content from `HEAD`. Not sure how something like this would be common though.

Reply:
FTR; A working tree is never a part of a repository. No git commands can "fetch" (or get in any way) a remote working tree. One repository can also be referenced from multiple working trees, each with their own index (stage).
It's only `gitk` / `tig` that propagates this myth via the "Uncommited changes", making them seem like a commit.

@psss For maximum correctness, the remote `ref` should default to `HEAD`, not `master`, as a remote `HEAD` points to what people (and github) usually call a "default branch".

Reply:
Local filesystem example desription updated to be explicit about the branches. @t184256, does it now look ok to you?

Reply:
for me it looks good

Reply:
@psss, IDK, now @comps made me wonder whether implicit HEAD would fare better than master or introduce even more confusion. But my original complaint is definitely resolved now, as it's clear it will clone from it and not use the working tree.